runtime/auth/signing/aws-signing-common/common/src/aws/smithy/kotlin/runtime/auth/signing/awssigning/common/AwsSigningResult.kt

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+package aws.smithy.kotlin.runtime.auth.signing.awssigning.common
+
+import aws.smithy.kotlin.runtime.http.request.HttpRequest
+
+/**
+ * Contains the result of AWS signing. Depending on the signing configuration, not all members may be assigned and some
+ * members (e.e., [signature]) may have a variable format.
+ */
+data class AwsSigningResult(
+    /**
+     * The signed HTTP request from the result, may be null if an HTTP request was not signed.
+     */
+    val signedRequest: HttpRequest?,
```

Comment:
I copied this [from **aws-crt-kotlin**](https://github.com/awslabs/aws-crt-kotlin/blob/main/src/common/src/aws/sdk/kotlin/crt/auth/signing/AwsSigningResult.kt) and only saw the **aws-sdk-kotlin** version later.

Honestly, this makes more sense to me since the signing result type is tied to the canonical signing operations defined in `AwsSigner`. The event signing is added on via extension methods and feels like it lives in a layer above the signer. Given that the only non-`HttpRequest` instance of the generic type param was `Message` and that was only used in the event stream signing code, it seemed better to [create an explicit type there](https://github.com/awslabs/aws-sdk-kotlin/blob/1a21a4773eb9380701e49997a82ae24786138a40/aws-runtime/protocols/aws-event-stream/common/src/aws/sdk/kotlin/runtime/protocol/eventstream/EventStreamSigning.kt#L81-L99).

If you feel strongly (or if there are future usecases for non-`HttpRequest` results of which I'm unaware) I can change it.

Reply:
> If you feel strongly (or if there are future usecases for non-HttpRequest results of which I'm unaware) I can change it.
I think it's more about correctness and extensibility. Nothing per-say ties signing to `HttpRequest`, it feels limiting to make it a nullable parameter when we have the language constructs to be explicit about what the result is.